test/test_id_model.cpp

```diff
@@ -1094,9 +1416,60 @@ TEST_F(IdModelTest, LoopPromotion8) {
 
   checkStep2Results(
       &fusion,
-      iel_graph,
+      tester.iel_graph,
       tester.idGraph(IdMappingMode::EXACT),
-      iel_promotion_map);
+      tester.s2_iel_promotion_map);
+
+  // Check Step 3 results. See the design doc for the expected results
+  std::vector<std::pair<std::unordered_set<Val*>, IterDomain*>>
+      s3_reference_map = {
+          // 1, 2, 3, 20, 6, 7, 17, 8, 9, 26, 14, 15, 29
```

Comment:
```suggestion
          // 1, 2, 3, 20, 6, 7, 17, 8, 9, 26, 14, 15, 29 -> 29
```

Reply:
Also, the doc for this needs to be updated. The arrow for this group is missing in the doc.